src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1089,6 +1089,24 @@
   <data name="DoNotAddArchiveItemPathToTheTargetFileSystemPathMessage" xml:space="preserve">
     <value>When creating path for '{0} in method {1}' from relative archive item path to extract file and the source is an untrusted zip archive, make sure to sanitize relative archive item path '{2} in method {3}'</value>
   </data>
+  <data name="DoNotCallEnumerableCastThatWillFailDescription" xml:space="preserve">
+    <value>If the source sequence contains any elements, the sequence returned by Cast will throw InvalidCastException at runtime when enumerated. Change the type(s) to those that are compatible or remove the call.</value>
+  </data>
+  <data name="DoNotCallEnumerableCastThatWillFailMessage" xml:space="preserve">
+    <value>Enumerable.Cast&lt;T&gt; will '{0}' can not be castetype '{1}' will fail</value>
```

Comment:
What about `Type '{0}' cannot be converted to '{1}' and will lead to a runtime failure`?

Reply:
Is 'converted' the correct terminology?  How about `Type '{0}' is incompatible with type '{1}' and will cause an exception at runtime`?

Reply:
From the doc page they say:

> If an element cannot be converted to type `TResult`, this method throws a `InvalidCastException`.